src/target/puya.c

```diff
@@ -114,39 +119,52 @@ bool puya_probe(target_s *target)
 	size_t flash_size = 0U;
 
 	const uint32_t dbg_idcode = target_mem32_read32(target, PUYA_DBG_IDCODE);
-	if ((dbg_idcode & 0xfffU) == 0) {
-		const uint32_t flash_ram_sz = target_mem32_read32(target, PUYA_FLASH_RAM_SZ);
+	if (dbg_idcode == 0x06188061U) {
+		const uint32_t flash_ram_sz = target_mem32_read32(target, PUYA_07X_FLASH_RAM_SZ);
+		flash_size = (((flash_ram_sz >> PUYA_FLASH_SZ_SHIFT) & PUYA_FLASH_SZ_MASK) + 1) << PUYA_FLASH_UNIT_SHIFT;
+		ram_size = (((flash_ram_sz >> PUYA_RAM_SZ_SHIFT) & PUYA_RAM_SZ_MASK) + 1) << PUYA_RAM_UNIT_SHIFT;
+		target->driver = "PY32F07x";
+		target_add_ram32(target, PUYA_RAM_START, 0x4000);
```

Comment:
Please name the size constant here, and mark it unsigned. We suggest something like `PUYA_07X_RAM_SIZE` to fit with the nomenclature being established here.